### PR TITLE
Ft add MVC, security & documentation features

### PR DESCRIPTION
**Pull request by bryan936M**
#### What does this PR do?
this creates the entire backend REST API
#### Description of Task to be completed
- Created the MVC structure
- Created the APIs for posts and messages.
- Added authentication to the REST API
#### How should this be manually tested?
- Download the project 
- Open the project in vscode
- Install the necessary dependencies (command: npm install)
- Start the server (command: npm run dev)
- Open the documentation page in the browser (url: http://localhost:3500/api-docs/)
#### Any background context you want to provide?
- Next task will be testing
#### What are the relevant pivotal tracker stories?
- Look into RE-CREATE BACKEND in trello(link: https://trello.com/invite/b/nLBA9kER/ATTI7620fcbf2169c3045a61a5c97751ff38B437BE1C/andela-portfolio-project)
#### Screenshots (if appropriate)

### CODE REVIEW

File: models/Comment.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+
+const commentSchema = new Schema({
+    : {
+        type: String,
+        required: true
```
Comment (chanceDira) [inline models/Comment.js]: Can you add field name here, 
e.g: 
comment: {
   type: String,
   required: true
}

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good Job @M936-B, just one change

File: models/Comment.js (hunk 1 above)
Comment (bryan936M) [reply models/Comment.js]: I know of this. I just want to leave it blank for now. Until I add comment section to front-end later. If that's okay,

Comment (chanceDira) [reply models/Comment.js]: I think you can have these fields here then, on the frontend you will put the titles you have in the comment model

**Pull request merged**